presto-docs/src/main/sphinx/security/server.rst

```diff
@@ -0,0 +1,254 @@
+coordinator from the username that is responsible for running queries. When
+running the Presto CLI, for example, the Presto username can be specified using
+the --user option.
+
+By default, the Presto coordinator allows any principal to run queries as any
+Presto user. In a secure environment, this is probably not desirable behavior.
+
+This behavior can be customized by implementing the
+``SystemAccessControlFactory`` and ``SystemAccessControl`` interfaces.
+
+``SystemAccessControlFactory`` is responsible for creating a
+``SystemAccessControl`` instance. It also defines a ``SystemAccessControl``
+name which is used by the administrator in a Presto configuration.
+
+``SystemAccessControl`` implementations have two responsibilities:
+ * Verifying whether or not a given Kerberos principal is authorized to execute
+   queries as a specific user.
+ * Determining whether or not a given user can alter values for a given system
+   property.
+
+The implementation of ``SystemAccessControl`` and
+``SystemAccessControlFactory`` must be wrapped as a plugin and installed on the
+Presto cluster.
+
+Presto Coordinator Node Configuration
+-------------------------------------
+
+config.properties
+^^^^^^^^^^^^^^^^^
+
+Kerberos authentication is configured in the coordinator node's
+config.properties file. The entries that need to be added are listed below.
+
+.. code-block:: none
+
+  http.server.authentication.enabled=true
+
+  http.server.authentication.krb5.service-name=presto
+  http.server.authentication.krb5.keytab=/etc/presto/presto.keytab
+  http.authentication.krb5.config=/etc/krb5.conf
+
+  http-server.https.enabled=true
+  http-server.https.port=7778
+
+  http-server.https.keystore.path=/etc/presto_keystore.jks
+  http-server.https.keystore.key=keystore_password
+
+* ``http.server.authentication.enabled``: Defines that the Presto coordinator should
+  enable authentication. Must be set to ``true``.
+* ``http.server.authentication.krb5.service-name``: Defines the Kerberos server
+  name for the Presto coordinator. Must match the Kerberos principal.
+* ``http.server.authentication.krb5.keytab``: The location of the keytab that
+  can be used to authenticate the Kerberos principal specified in
+  ``http.server.authentication.krb5.service-name``.
+* ``http.authentication.krb5.config``: The location of the Kerberos
+  configuration file.
+* ``http-server.https.enabled``: Specifies if Presto coordinator should listen
+  for HTTPS.
+* ``http-server.https.port``: HTTPS server port.
+* ``http-server.https.keystore.path``: The location of the Java Keystore file
+  that will be used to secure TLS.
+* ``http-server.https.keystore.key``: The password for the keystore. This must
+  match the password you specified when creating the keystore.
+
+access-controls.properties
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Once a plugin that implements ``SystemAccessControl`` and
+``SystemAccessControlFactory`` has been installed, it is configured using an
```

Comment:
A few questions: Is the plugin installed on the node? Is it installed on a worker or Presto coordinator node, or both?